src/api/api.js

```diff
@@ -18,7 +18,7 @@ export const mapData = ({ facts = {}, ...oneResult }) => ({
     ...oneResult,
     rawFacts: facts,
     facts: {
-        ...facts.reduce((acc, curr) => ({ ...acc, [curr.namespace]: curr.facts }), {}),
+        ...facts.reduce?.((acc, curr) => ({ ...acc, [curr.namespace]: curr.facts }), {}),
```

Comment:
I assume that the mock for `'/api/inventory/v1/hosts/test-id/system_profile?fields%5Bsystem_profile%5D%5B%5D=operating_system'` is not right and something else is passed in, not sure yet what to properly pass in as a mock result.